lib/game.rb

```diff
@@ -75,118 +78,27 @@ def change_current_player
     @players.rotate!
   end
 
-  def is_valid?(input)
-    is_valid_column?(input) && !is_column_full?(input)
-  end
-  
-  def is_valid_column?(input)
-    %w[A B C D E F G].include?(input)
-  end
-  
-  def is_column_full?(input)
-    @board.visual[input].none?(".")
-  end
-
   def game_over
     if check_for_draw?
       puts @message_bot.draw
       sleep(5)
-      @board = Board.new
-      @turns_taken = 0
+      reset_game
       start
     elsif win?
       puts @message_bot.win(current_player)
       sleep(5)
-      @board = Board.new
-      @turns_taken = 0
+      reset_game
       start
     end
   end
 
-  def check_for_draw?
-    @turns_taken > 42
+  def reset_game
```

Comment:
I like separating this out into a helper method.

Reply:
I also like creating the 'reset_game' method - nice work!